src/libpmempool/replica.c

```diff
@@ -849,68 +878,105 @@ check_replica_sizes(struct pool_set *set, struct poolset_health_status *set_hs)
  */
 int
 replica_check_poolset_health(struct pool_set *set,
-		struct poolset_health_status **set_hsp, unsigned flags)
+		struct poolset_health_status **set_hsp, unsigned flags,
+		PMEM_progress_cb progress_cb)
 {
 	LOG(3, "set %p, set_hsp %p, flags %u", set, set_hsp, flags);
+
+	char *msg = "Checking poolset health";
+	size_t progress_max = 11;
```

Comment:
If it's not too much trouble, it would look nicer if you encapsulated the progress steps of each individual operation in an enum - it would certainly increase clarity of the code and make it less error prone to modify.